elasticsearch/es_client_v7.go

```diff
@@ -301,3 +302,85 @@ func (es *ESClientV7) GetTotalDiskUsage(ctx context.Context) (string, error) {
 
 	return totalDiskUsage, nil
 }
+
+func (es *ESClientV7) getDBUserRole(ctx context.Context) (error, bool) {
+	req := esapi.SecurityGetRoleRequest{
+		Name: []string{CustomUser},
+	}
+	res, err := req.Do(ctx, es.client.Transport)
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+			klog.Errorf("failed to close response body from GetDBUserRole", err)
+		}
+	}(res.Body)
+
+	if err != nil {
+		fmt.Println("faced error while making request in GetDBUserRole function")
+		return err, false
+	}
+	if res.IsError() {
+		return err, false
+	}
+
+	return nil, true
+}
+
+func (es *ESClientV7) EnsureDBUserRole(ctx context.Context) error {
+	err, flg := es.getDBUserRole(ctx)
+
+	if err != nil {
+		return err
+	}
+	if !flg {
+		dbPrivileges := map[string]interface{}{
+			Names:                  []string{All},
+			Privileges:             []string{PrivilegeReadKey, PrivilegeWriteKey, PrivilegeCreateIndexKey},
+			AllowRestrictedIndices: false,
+		}
+		applicationPrivileges := map[string]interface{}{
+			Application: Kibana,
+			Privileges:  []string{PrivilegeReadKey, PrivilegeWriteKey},
+			Resources:   []string{Any},
+		}
+		transientMetaPrivileges := map[string]interface{}{
+			Enabled: true,
+		}
+		userRoleReqMap := map[string]interface{}{
+			Cluster:           []string{All},
+			Indices:           []map[string]interface{}{dbPrivileges},
+			Applications:      []map[string]interface{}{applicationPrivileges},
+			RunAs:             []string{},
+			TransientMetadata: transientMetaPrivileges,
+		}
```

Comment:
Use a struct to built the role request body

Reply:
Example:

type UserRoleReqMap struct {
	Cluster []string `json:cluster`
        DBPriviledges DBPriviledges `json:`
}
type DBPriviledges struct {
	Names []string `json:names`
	AllowRestrictedIndices bool `json:allow_restricted_indices`
}